### DISCUSSION TITLE
对多窗口的 devtool 唤起进行分离

### DISCUSSION THREAD

**Discussion by BTMuli** [vuejs/devtools#145] [Q&A]
对于多窗口应用，存在页面 A、B、C的情况下，页面A唤起，页面B、C也会同时唤起。

希望能对多窗口进行处理。

**Comment by alexzhang1030** [comment]
你可以使用 devtools 单页面来解决这个问题
<img width="1094" alt="image" src="https://github.com/vuejs/devtools-next/assets/49969959/a5638f6d-afc1-4892-a7b5-f0cacc8e02ba">

**Comment by BTMuli** [comment]
> 你可以使用 devtools 单页面来解决这个问题

这个页面一直显示 `Waiting for connection...`，我用的是 `vite-plugin-dev-tools` 配置的，需要加什么其他配置吗

**Comment by alexzhang1030** [comment]
需要你打开一次页面

**Comment by BTMuli** [comment]
> 需要你打开一次页面

但是我用的是 tauri 框架，直接打开 url 并不能正确渲染
![image](https://github.com/vuejs/devtools-next/assets/72692909/ea9d66e1-3931-435f-b81c-f243c8e583fd)

**Comment by alexzhang1030** [comment]
tauri 理论上打开一次 app 然后在浏览器上单独浏览 devtools 应该也可以，打开页面主要是为了请求一次 assets 让 vite 处理然后让 devtools 接收数据

**Comment by alexzhang1030** [comment]
刚刚测试了一下，单独窗口的通信目前还只能依赖同源窗口，对于 tauri 和 electron 等还存在这样的问题。你可以用 `standalone` mode, 后续会引入一定的配置来让 devtools 的 open 状态不依赖 storage

**Comment by BTMuli** [comment]
> 你可以用 `standalone` mode

那就是不使用 vite-plugin 而是自己配置 vue-devtools  是吗

**Reply by alexzhang1030** [reply]
是 CLI，可以查看 [官网](https://devtools-next.vuejs.org/guide/standalone)

这种方式是采用另一种通信的，所以没办法直接使用 vite 相关的功能（graph, inspect, go to editor 等)

**Reply by BTMuli** [reply]
7.0.x 版本的通过 cli 调用不了，而 6.5.1 的是正常的，有没有什么 debug 命令可以查看报错的

**Reply by alexzhang1030** [reply]
7.x 是当前这个 repo 的代码，<7.0.0 是 [github.com/vuejs/devtools](https://github.com/vuejs/devtools) 这个 repo 的代码。我之前测试 是可以启动的，有什么额外的信息吗

**Reply by BTMuli** [reply]
任务栏能看到 electron 进程，但是终端输入后无响应，除此之外没有其他信息了😶

**Reply by alexzhang1030** [reply]
https://github.com/vuejs/devtools-next/assets/49969959/59d5cb84-d88c-43b1-b36b-8ed35ac56658

效果是这样的, 不过 electron 版本是相当早期版本，会有一些问题，如果你升级到最新版本并尝试多次无法唤起后，可以提一个 issue 了
